soql-analyzer/src/main/scala/com/socrata/soql/functions/MonomorphicFunction.scala

```diff
@@ -6,7 +6,7 @@ case class MonomorphicFunction[Type](function: Function[Type], bindings: Map[Str
   def this(identity: String, name: FunctionName, parameters: Seq[Type], repeated: Seq[Type], result: Type, isAggregate: Boolean = false) =
     this(Function(identity, name, Map.empty, parameters.map(FixedType(_)), repeated.map(FixedType(_)), FixedType(result), isAggregate), Map.empty)
 
-  require(bindings.keySet == function.parameters.collect { case VariableType(n) => n }.toSet, "bindings do not match")
+  require(bindings.keySet.filterNot(_ == "?") == function.parameters.collect { case VariableType(n) => n }.toSet, "bindings do not match")
```

Comment:
It'd probably be better to create a new subclass of `TypeLike` rather than treating `VariableType("?")` specially

Reply:
Thought about that too.  Yeah, that is better.